src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -598,6 +598,16 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._userNameInputBox.enable();
 			this._passwordInputBox.enable();
 			this._rememberPasswordCheckBox.enabled = true;
+
+			const recentConnections = this._connectionManagementService.getRecentConnections();
+			if (recentConnections && recentConnections.length > 0) {
+				const self = this;
```

Comment:
nit: You shouldn't need this - as long as you use a lambda capture below it'll capture the this context correctly.